common/workunit/workunit.cpp

```diff
@@ -331,7 +331,10 @@ class CConstGraphProgress : public CInterface, implements IConstWUGraphProgress
                 throwUnexpected();
             }
 
-            IPropertyTree * next = curTarget->addPropTree(tag);
+            VStringBuffer xpath("%s[@id=\"%s\"]", tag, id);
+            IPropertyTree * next = curTarget->queryPropTree(xpath);
+            if (!next)
+                next = curTarget->addPropTree(tag);
             next->setProp("@id", id);
```

Comment:
This line could be inside the if() I think?

Reply:
true, I will change